htdocs/core/lib/functions.lib.php

```diff
@@ -9231,3 +9231,42 @@ function readfileLowMemory($fullpath_original_file_osencoded, $method = -1)
 		fclose($handle2);
 	}
 }
+
+/**
+ * Create a button to copy $valuetoprint in the clipboard
+ * @param string $valuetoprint the value to print
+ * @param string $spanid unique id for a span
+ * @param Translate $langs the $langs variable
+ * @return string the string to print for the button
+ */
+function showValueWithCopyAndPasteButton($valuetoprint, $spanid, $langs)
+{
+	$stringtoprint= '<span id="'.$spanid.'">';
+	$stringtoprint.= $valuetoprint .'</span>&ensp;';
+	$stringtoprint.= '</span>';
+	$stringtoprint.='<span class="opacitymedium"  id="'.dol_escape_json($spanid).'clipboardChar" onclick="'.dol_escape_json($spanid).'clipBoardCopy()"><i " class="far fa-clipboard"></i></span>';
```

Comment:
To keep a good separation between HTML and javascript and keep javascript no being intrusive into HTML code, take a look a Jquery that is enabled by default in dolibarr.
You should be able to replace the inline onclick tag by javascript code all inside the <script> section.
```
jQuery("#".dol_escape_json($spanid).'clipboardChar').click(function() {
    clipBoardCopy();
})
```

Code is really easier to maintain when there is a 100% separation between HTML and javascript.